common/utils/has-own-property.ts

```diff
@@ -1,6 +0,0 @@
-export default function hasOwnProperty<
```

Comment:
Just to confirm; this just wasn't used anywhere so we're getting rid of it?

Reply:
Ah yup, I was going to consolidate it into type-guards but it wasn't used anywhere so I binned it.